libgalois/include/katana/PropertyGraph.h

```diff
@@ -155,23 +155,30 @@ class KATANA_EXPORT GraphTopology {
 /// comprise the physical representation of the logical property graph.
 class KATANA_EXPORT PropertyGraph {
 public:
-  /// TypeSetID uniquely identifies/contains a combination/set of types
-  /// TypeSetID is represented using 8 bits
-  /// TypeSetID for nodes is distinct from TypeSetID for edges
-  using TypeSetID = uint8_t;
-  static constexpr TypeSetID kUnknownType = TypeSetID{0};
-  static constexpr TypeSetID kInvalidType =
-      std::numeric_limits<TypeSetID>::max();
-  /// A set of TypeSetIDs
-  using SetOfTypeSetIDs =
-      std::bitset<std::numeric_limits<TypeSetID>::max() + 1>;
-  /// A set of type names
+  /// EntityTypeID uniquely identifies an entity (node or edge) type
+  /// EntityTypeID for nodes is distinct from EntityTypeID for edges
+  /// This type may either be an atomic type or an intersection of atomic types
+  /// EntityTypeID is represented using 8 bits
+  using EntityTypeID = uint8_t;
+  static constexpr EntityTypeID kUnknownType = EntityTypeID{0};
+  static constexpr EntityTypeID kInvalidType =
+      std::numeric_limits<EntityTypeID>::max();
+  /// A set of EntityTypeIDs
+  using SetOfEntityTypeIDs =
+      std::bitset<std::numeric_limits<EntityTypeID>::max() + 1>;
+  /// A set of names of atomic types
   using SetOfTypeNames = std::unordered_set<std::string>;
-  /// A map from TypeSetID to the set of the type names it contains
-  using TypeSetIDToSetOfTypeNamesMap = std::vector<SetOfTypeNames>;
-  /// A map from the type name to the set of the TypeSetIDs that contain it
-  using TypeNameToSetOfTypeSetIDsMap =
-      std::unordered_map<std::string, SetOfTypeSetIDs>;
+  using TypeNameIterator = std::unordered_set<std::string>::const_iterator;
+  using TypeNamesRange = StandardRange<TypeNameIterator>;
+  /// A map from EntityTypeID to the set of names of atomic types it intersects
+  using EntityTypeIDToSetOfTypeNamesMap = std::vector<SetOfTypeNames>;
+  /// A map from the type name to the set of the EntityTypeIDs that intersect it
+  using TypeNameToSetOfEntityTypeIDsMap =
+      std::unordered_map<std::string, SetOfEntityTypeIDs>;
```

Comment:
You seem to be adding "atomic" to specify which types you are naming in some places, but not others. Is there a reason?

Reply:
Yes, some of these are only applicable for atomic types.